deployment/cdk/opensearch-service-migration/lib/network-stack.ts

```diff
@@ -9,13 +9,38 @@ import {StackPropsExt} from "./stack-composer";
 import {StringParameter} from "aws-cdk-lib/aws-ssm";
 
 export interface NetworkStackProps extends StackPropsExt {
-    readonly vpcId?: string
-    readonly availabilityZoneCount?: number
+    readonly vpcId?: string,
+    readonly availabilityZoneCount?: number,
+    readonly targetClusterEndpoint?: string
 }
 
 export class NetworkStack extends Stack {
     public readonly vpc: IVpc;
 
+    // Validate a proper url string is provided and return an url string which contains a protocol, host name, and port.
+    // If a port is not provided, the default protocol port (e.g. 443, 80) will be explicitly added
+    static validateAndReturnFormattedHttpURL(urlString: string) {
+        // URL will throw error if the urlString is invalid
+        let url = new URL(urlString);
+        if (url.protocol !== "http:" && url.protocol !== "https:") {
```

Comment:
the colon is included in the protocol field?

Reply:
For this typescript library yes, i was a little taken as well